### ISSUE TITLE
Extension handlers only handle extensions with "object" type.

### ISSUE THREAD

**Issue by timburks** [googleapis/gnostic#34]
While testing extensions with a real-world schema, I discovered that extensions of array and scalar types were not being handled. Here's a schema that illustrates the problem:

```
title: "JSON schema for apis.guru extensions"
$schema: "http://json-schema.org/draft-04/schema#"
type: object
definitions:
  Logo:
    type: object
    id: x-logo
    required:
    - url
    properties:
      url:
        type: string
        format: url
  APIClientRegistration:
    type: object
    id: x-apiClientRegistration
    required:
    - url
    properties:
      url:
        type: string
        format: url
  Origin:
    id: x-origin
    type: object
    required:
    - format
    - url
    - version
    properties:
      format:
        type: string
      url:
        type: string
        format: url
      version:
        type: string
  Preferred:
    id: x-preferred
    type: boolean
  ProviderName:
    id: x-providerName
    type: string
  ServiceName:
    id: x-serviceName
    type: string
  APISGuruCategories:
    id: x-apisguru-categories
    type: array
  Tags:
    id: x-tags
    type: array
  UnofficialSpec:
    id: x-unofficialSpec
    type: boolean
```

**Comment by timburks** [comment]
[Here](https://github.com/APIs-guru/openapi-directory/wiki/specification-extensions) is a good list of extension sets that would provide some good validation for gnostic extensions.

**Comment by guptasu** [comment]
Issue acknowledged, will take a look. Currently swamped with other tasks. 

Resolving this is going to be tricky because google.protobuf.Any type can only represent Message types. Since we are using google.protobuf.Any to represent extensions, we will need to use [wrapper types](https://github.com/google/protobuf/blob/master/src/google/protobuf/wrappers.proto) to convert primitives into google.protobuf.Any when reading an extension of type int/string etc. This can cause usability issue in the extension handers where author of the extension hander will have to unwrap the wrappertype by doing something like in java : 

StringValue s = .. read string value from Any Type..
string extensionString = s.value.

need to dig more on to understand if this is the only route we have.

**Comment by MikeRalphson** [comment]
If you filter [this list](https://github.com/Mermade/openapi-specification-extensions/blob/master/extensions/combined.tsv) by object you can find more examples.

**Comment by timburks** [comment]
@guptasu Something like a wrapper type SGTM. We should mock up a few samples to be sure. 

@MikeRalphson Thanks, that's fantastic!

**Comment by timburks** [comment]
Resolved in [PR #44](https://github.com/googleapis/gnostic/pull/44).
